docs/usage/two-phase-transfer.md

```diff
@@ -0,0 +1,67 @@
+# Two-phase Transfer
+
+The name "two-phase transfer" is a reference to the [two-phase commit
+protocol for distributed
+transactions](https://en.wikipedia.org/wiki/Two-phase_commit_protocol).
+
+Single-phase transfers post funds to accounts immediately when they
+are created. That is, they affect the `debits_posted` and
+`credits_posted` fields on relevant accounts.
+
+In contrast to single-phase transfers, a two-phase transfer moves
+funds in stages:
+
+1. First, the pending transfer reserves funds. While reserved, they
+cannot be used by either the payer or payee. Only the `debits_pending`
+and `credits_pending` fields are modified on the relevant accounts at
+this point.
+
+2. Later, the application creates another transfer — with flags either
+marked `post_pending_transfer` or `void_pending_transfer`. The former
+moves all (or part) of the reserved funds to the destination; the
+latter reverts them to the original account.
+
+## `credits_must_not_exceed_debits` and `debits_must_not_exceed_credits`
+
+The pending transfer's amount is reserved in a way that the second
+step in a two-phase transfer will never cause the accounts' configured
+balance invariants to be broken, whether the second step is a post or
+void.
+
+### Pessimistic pending transfers
+
+If an account with `debits_must_not_exceed_credits` has
+`credits_posted = 100` and `debits_posted = 70` and a pending transfer
+is started causing the account to have `debits_pending = 50`, the
+*pending* transfer will fail. It will not wait to get to *posted*
+status to fail.
+
+## All transfers are immutable
+
+To reiterate, completing a two-phase transfer (by either marking it
+void or posted) does not involve modifying a transfer. Instead you
+create a new transfer.
+
+The first transfer that is marked pending will always have its pending
+flag set.
+
+The second transfer will have a `post_pending_transfer` or
+`void_pending_transfer` flag set and a `pending_id` field set to the
+ID of the first transfer. The ID of the second transfer will be
```

Comment:
Instead of `ID`, what about `id`? (That makes it more clear that we aren't informally referring to the `pending_id`, which was just mentioned.)

Reply:
Good idea, adjusted.